editor: allows user to select infra

jacomyal opened the pull request:
This commit closes #1986.

Details:
- Adds a button in the right of the map that opens a modal where the user can select which infra to edit
- Also removes viewport data from URLs in editor, since they caused more trouble than they solved issues

jacomyal committed:
editor: allows user to select infra

This commit fixes #1986.

Details:
- Adds a button in the right of the map that opens a modal where the
user can select which infra to edit
- Also removes viewport data from URLs in editor, since they caused more
trouble than they solved issues

sim51 commented:
When I have a selection and I switch to an other infrastructure, the selection is still there with the data of the previous infra

jacomyal committed:
editor: flushes editor state on infra change

This commit is related to #1986. It fixes an issue raised by @sim51 in a
comment to PR #2069:

> When I have a selection and I switch to an other infrastructure, the
selection is still there with the data of the previous infra

Details:
- Groups activeTool and toolState in a single useState call in
Editor.tsx internal logic, to prevent some invalid app state
- Adds a new `reset` editor action to reset editor state (except for
schema, that never changed for now)
- Flushes editor and tool state (and active tool, actually) when the
infra changes
